android/src/main/java/nl/fhict/happynews/android/activity/SourcesSettingsActivity.java

```diff
@@ -28,16 +32,26 @@ protected void onCreate(Bundle savedInstanceState) {
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         setTitle(R.string.title_settings_sources);
 
-        preferences = getApplicationContext().getSharedPreferences(
-            getString(R.string.preference_file_key), Context.MODE_PRIVATE);
-
         sourcesListView = (ListView) findViewById(R.id.sourcesListView);
-        sources = createSourcesObjects();
+        sources = SourceController.getInstance().getSources(getApplicationContext());
+        Collections.sort((List<SourceSetting>)sources);
```

Comment:
Casting `Collection` to `List` feels a bit scary. If it is always a `List` we might as well make `SourceController#getSources()` return `List`.